### PR TITLE
HWS: Support CX8 HW actions field changes

### PR DESCRIPTION
**Pull request by hamdanigbaria**

### CODE REVIEW

File: hws/src/dr_common_functions.py [hunk 1]
```diff
@@ -137,20 +137,22 @@ def dr_parse_add_action(action_dw_0, action_dw_1, parse_value=True):
...
 def dr_parse_remove_by_size_action(action_dw_0, action_dw_1):
     action = {"type" : "Remove by size"}
-    start_anchor = int(action_dw_0[10 : 16], 2)
+    cx8 = True if _config_args.get("cx8") else False
+    start_anchor = int(action_dw_0[8 : 15], 2) if cx8 else int(action_dw_0[10 : 16], 2)
     field = modify_pattern_anchor_dic.get(start_anchor)
     action["start_anchor"] = field if field != None else start_anchor
     action["outer_l4_removed"] = int(action_dw_0[16 : 17], 2)
-    action["start_offset"] = int(action_dw_0[18 : 25], 2)
+    action["start_offset"] = int(action_dw_0[15 : 23], 2) if cx8 else int(action_dw_0[18 : 25], 2)
```
Comment (alexvesker) [inline hws/src/dr_common_functions.py:145]: maybe cleaner to have one
```
if cx8: 
   ...
else
   ...
```

File: hws/src/dr_common_functions.py [hunk 2]
```diff
@@ -137,20 +137,22 @@ def dr_parse_add_action(action_dw_0, action_dw_1, parse_value=True):
...
     action["size"] = int(action_dw_0[26 : 32], 2)
 
     return action
 
 
 def dr_parse_remove_header2header_action(action_dw_0, action_dw_1):
     action = {"type" : "Remove header2header"}
-    start_anchor = int(action_dw_0[10 : 16], 2)
-    end_anchor = int(action_dw_0[18 : 24], 2)
+    cx8 = True if _config_args.get("cx8") else False
```
Comment (alexvesker) [inline hws/src/dr_common_functions.py:153]: so total diff is
4 actions?
dr_parse_remove_by_size_action
dr_parse_remove_header2header_action
dr_parse_insert_inline_action
dr_parse_insert_by_pointer_action

future, if we will have one more device and more diffs these if`s will grow,
you will need a profile/config per device,

Comment (hamdanigbaria) [reply hws/src/dr_common_functions.py:153]: yes
if need more changes in the future will change the whole logic

File: hws/src/dr_common_functions.py (hunk 1 above)
Comment (hamdanigbaria) [reply hws/src/dr_common_functions.py]: Done

File: hws/src/dr_common_functions.py [hunk 3]
```diff
@@ -164,11 +179,20 @@ def dr_parse_remove_header2header_action(action_dw_0, action_dw_1):
...
-    start_anchor = int(action_dw_0[10 : 16], 2)
+    start_anchor = 0
+    start_offset = 0
+
+    if _config_args.get("cx8"):
+        start_anchor = int(action_dw_0[8 : 15], 2)
+        start_offset = int(action_dw_0[15 : 23], 2)
+    else:
+        start_anchor = int(action_dw_0[10 : 16], 2)
+        start_offset = int(action_dw_0[16 : 23], 2)
```
Comment (kliteyn) [inline hws/src/dr_common_functions.py:190]: The original code says action_dw_0[18 : 23], not [16 : 23].
Which one is the correct one?

File: hws/src/dr_common_functions.py [hunk 4]
```diff
@@ -178,11 +202,20 @@ def dr_parse_insert_inline_action(action_dw_0, action_dw_1, parse_value=True):
...
```
Comment (kliteyn) [inline hws/src/dr_common_functions.py:213]: same here - it was [18 : 23]

**Review by kliteyn — COMMENTED** [review]
please see my comments

File: hws/src/dr_common_functions.py (hunk 3 above)
Comment (hamdanigbaria) [reply hws/src/dr_common_functions.py:190]: it should be like the new one

File: hws/src/dr_common_functions.py (hunk 4 above)
Comment (hamdanigbaria) [reply hws/src/dr_common_functions.py:213]: it should be like the new one

**Review by kliteyn — APPROVED** [review]

**Pull request merged**